src/wasm-type.h

```diff
@@ -50,16 +54,17 @@ class Type {
   Type() = default;
 
   // ValueType can be implicitly upgraded to Type
-  constexpr Type(ValueType id) : id(id){};
+  constexpr Type(ValueType id) : id(id), _size(id == none ? 0 : 1){};
 
   // But converting raw uint32_t is more dangerous, so make it explicit
-  constexpr explicit Type(uint32_t id) : id(id){};
+  constexpr explicit Type(uint32_t id) : id(id), _size(unknownSize){};
```

Comment:
why is the size unknown here, mustn't the id be valid? if so shouldn't we get the size from the caller too?

Reply:
I wanted to avoid taking the lock to look up the size, but idk how important that is. I'll investigate further.

Reply:
It turns out that this constructor is primarily used in the C API to convert from `BinaryenType` aka `uint32_t` to a Type. Right now `BinaryenType` values are type IDs, but we could pack the size into them as well. The downside of that would be that it would make the conversion from `BinaryenType` to `Type` even more dangerous and error prone and generally add a maintenance burden to the C API. I think it's probably better to take the lock and look up the size in the constructor.